rasa/shared/core/training_data/story_reader/yaml_story_reader.py

```diff
@@ -338,26 +338,47 @@ def _validate_that_utterance_is_in_domain(self, utterance: UserUttered) -> None:
             )
 
     def _parse_or_statement(self, step: Dict[Text, Any]) -> None:
-        utterances = []
+        events: List = []
 
-        for utterance in step.get(KEY_OR):
-            if KEY_USER_INTENT in utterance.keys():
-                utterance = self._parse_raw_user_utterance(utterance)
+        for item in step.get(KEY_OR):
+            if KEY_USER_INTENT in item.keys():
+                utterance = self._parse_raw_user_utterance(item)
                 if utterance:
-                    utterances.append(utterance)
+                    events.append(utterance)
+            elif KEY_CHECKPOINT_SLOTS in item.keys():
+                if isinstance(item, dict):
+                    for key, value in item.items():
+                        parsed_events = self._parse_events(
+                            SlotSet.type_name, {key: value}
+                        )
+                        events.extend(parsed_events)
+                elif isinstance(item, str):
```

Comment:
Is it possible for `item` to be a str if this is nested under `elif KEY_CHECKPOINT_SLOTS in item.keys()`?